lib/backend/Memory.js

```diff
@@ -83,11 +88,18 @@ export default class Memory {
     * @return {undefined}
     */
     incr(key, cb) {
+        assert.strictEqual(typeof key, 'string');
         process.nextTick(() => {
             if (this.data[key] === undefined) {
                 this.data[key] = 0;
             }
-            return cb(null, this.data[key]++);
+            const val = parseInt(this.data[key], 10);
+            if (Number.isNaN(val)) {
+                throw new Error('Value at key cannot be represented as a ' +
```

Comment:
The list metrics api call ignores error from a command, logs them and moves on to the next. So if errors are sent in callbacks we would never catch them in the tests.  Since this Memory backend is intended purely for testing (unit/functional) I have decided to purposefully throw errors which would stop the test execution. It's the same reasoning for assertions, it's so that we would catch them in the tests.